airbyte-server/src/main/java/io/airbyte/server/handlers/JobHistoryHandler.java

```diff
@@ -56,24 +54,20 @@ public JobReadList listJobsFor(JobListRequestBody request) throws IOException {
     Preconditions.checkNotNull(request.getConfigTypes(), "configType cannot be null.");
     Preconditions.checkState(!request.getConfigTypes().isEmpty(), "Must include at least one configType.");
 
-    final List<JobConfig.ConfigType> configTypes = request.getConfigTypes()
+    final Set<ConfigType> configTypes = request.getConfigTypes()
         .stream()
         .map(type -> Enums.convertTo(type, JobConfig.ConfigType.class))
-        .collect(Collectors.toList());
+        .collect(Collectors.toSet());
     final String configId = request.getConfigId();
 
-    // get jobs for each type and merge them into a single list sorted by created at.
-    Iterable<JobWithAttemptsRead> jobReads = ImmutableList.of();
-    for (final JobConfig.ConfigType configType : configTypes) {
-      final List<JobWithAttemptsRead> jobReadsForType = jobPersistence.listJobs(configType, configId)
-          .stream()
-          .map(JobConverter::getJobWithAttemptsRead)
-          .collect(Collectors.toList());
-
-      jobReads = Iterables.mergeSorted(ImmutableList.of(jobReads, jobReadsForType), Comparator.comparing(v -> v.getJob().getCreatedAt()));
-    }
-
-    return new JobReadList().jobs(MoreStreams.toStream(jobReads).collect(Collectors.toList()));
+    final List<JobWithAttemptsRead> jobReads = jobPersistence.listJobs(configTypes,
+        configId,
+        request.getPagesize() != null ? request.getPagesize() : 25,
```

Comment:
we tend to try to avoid "magic" numbers. the convention we usually follow is to put a static constant at the top of the class (at least for source classes, in tests it is more at the developers discretion).

e.g. `private static final int DEFAULT_PAGE_SIZE = 25;`